Freesurfer reader MGH/MGZ reader and Triange surface (white)

bwagnerwork opened the pull request:
Still question on how to setup orientation and origin.  Especially with the dead center of volume reference of the surfaces.

NicolasRannou commented:
Awesome

Can you fix the styling issues reported in the build please?
https://travis-ci.org/FNNDSC/ami/builds/335677286?utm_source=github_status&utm_medium=notification

Running the following should fix most of them:
```
eslint src/**/*.js --fix
```

bwagnerwork commented:
Cleaned up except for some oddities around some errors I throw for the freesurfer surface types that have not been written.

/home/travis/build/FNNDSC/ami/src/loaders/loaders.freesurfer.js
  102:13   error    Unexpected lexical declaration in case block  no-case-declarations
  102:13   error    Unexpected lexical declaration in case block  no-case-declarations
  102:13   error    Unexpected lexical declaration in case block  no-case-declarations

bwagnerwork commented:
Fixed the eslint error and added the quad-face reader.  How would you go about specifying color/value files that should probably be read in at the same time as the surface?  For example,  adding a  cortical parcellation to a lh.pial?

NicolasRannou commented:
Nice!

Does the orientation work now by any chance?

NicolasRannou commented on src/parsers/parsers.js:
```diff
@@ -1,9 +1,11 @@
 import Dicom from './parsers.dicom';
 import Nifti from './parsers.nifti';
 import Nrrd from './parsers.nrrd';
+import Mgh from './parsers.mgh';
```
Should be beofre Nifti (alphabetical order)

NicolasRannou commented on src/parsers/parsers.js:
```diff
@@ -1,9 +1,11 @@
     Dicom as DicomParser,
     Nifti as NiftiParser,
     Nrrd as NrrdParser,
+    Mgh as MghParser,
```
Should be before Nifti (alphabetical order)

bwagnerwork commented:
Ahh, Orientation.  Nope.  MGH by itself is OK.  MGH with Freesurfer overlays not so much. Unfortunately that is my main use case.  I added a flag that can be set before reading the MGH/Freesurfer files (ParsersMgh.FREESURFER_ORIENT accessible via AMI.MghParser.FREESURFER_ORIENT) so that I can switch between the two methods.  If you get a chance to look at the sample data I uploaded and see a better way to do it I'd appreciate it.

NicolasRannou commented:
THanks for your help, will do now

NicolasRannou commented:
FYI I think the way to go is to display the volume at the "right" location, so it overlays nicely with other volume. The loader could also provide the transform to be applied to the FS mesh to overlay nicely somehow.

NicolasRannou commented:
@bwagnerwork I believed almost all the issues are fixed now - no more tricks hopefully.

The only requirement is to apply the following transform to the FS model, in order to match the volume:

```
         let RASToLPS = new THREE.Matrix4();
          const worldCenter = r1.stackHelper.stack.worldCenter();
          RASToLPS.set( -1, 0, 0, worldCenter.x,
                        0, -1, 0, worldCenter.y,
                        0, 0, 1, worldCenter.z,
                        0, 0, 0, 1);
          object.mesh.applyMatrix(RASToLPS);
          r0.scene.add(object.mesh);
```

Sill few things to figure out but minor things hopefully. The main problem was the the coordinate system was not right handed, so that is something that we need to detect in the MGH parser.

![screen shot 2018-02-09 at 4 34 14 pm](https://user-images.githubusercontent.com/214063/36035436-30344e3a-0db7-11e8-812d-92d117a9d6a1.png)

NicolasRannou commented:
Awesome thx a lot!

NicolasRannou commented:
(did merge with the fixes)

NicolasRannou commented:
Demo should be live in a bit at: https://fnndsc.github.io/ami/#viewers_quadview

Please let me know if that looks fine to you!
